test/services/users.test.ts

```diff
@@ -1,13 +1,245 @@
+      } catch (err: any) {
+        const {code, message} = err;
+        expect(code).to.equal(400);
+        expect(message).to.equal('ValidationError: "id" must be a valid GUID');
+      }
+    });
+  });
+
+  describe('find user', () => {
+
+    before(async () => {
+      const {id} = await service.create(userInfo);
+      userId = id;
+    });
+    after(async () => {
+      await service.remove(userId);
+    });
+
+    it('should succeed - find user', async () => {
+      const {total} = await service.find();
+      expect(total).not.to.equal(0);
+    });
+  });
+
+  describe('update (patch) user', () => {
+
+    before(async () => {
+      const {id} = await service.create(userInfo);
+      userId = id;
+    });
+    after(async () => {
+      await service.remove(userId);
+    });
+
+    it('should succeed - update(patch) user', async () => {
+      const { firstName, lastName } = await service.patch(userId, updateUser);
+      expect(firstName).to.equal(updateUser.firstName);
+      expect(lastName).to.equal(updateUser.lastName);
+    });
+
+    it('should succeed - user was updated(patch) successfully', async () => {
+      const { firstName, lastName } = await service.get(userId);
+      expect(firstName).to.equal(updateUser.firstName);
+      expect(lastName).to.equal(updateUser.lastName);
+    });
+
+    it('should fail - invalid user id', async () => {
+      try{
+        await service.patch(unexistedId,updateUser);
+        expect.fail('call should have failed');
+      } catch (err: any) {
+        const {code} = err;
+        expect(code).to.equal(404);
+      }
+    });
+
+    it('should fail - invalid user data', async () => {
+      const invalidUserData ={
+        firstName: 12,
+        lastName: 'updated user last name',
+        email: 123456,
+        password: 'supersecret'
+      };
+      try{
+        await service.patch(userId,invalidUserData);
+        expect.fail('call should have failed');
+      } catch (err: any) {
+        const {code, message} = err;
+        expect(code).to.equal(400);
+        expect(message).to.equal('Invalid data');
+      }
+    });
+  });
+
+  describe('delete user', () => {
+
+    it('should fail - id not found', async () => {
+      try{
+        await service.remove(unexistedId);
+        expect.fail('call should have failed');
+      } catch (err: any) {
+        const {code, message} = err;
+        expect(code).to.equal(404);
+        expect(message).to.equal('No record found for id \'783deee0-3732-11ec-9301-13adbda06b66\'');
+      }
+    });
+
+    it('should fail - invalid id', async () => {
+      try{
+        await service.remove('123456', { provider: 'rest' });
+        expect.fail('call should have failed');
+      } catch (err: any) {
+        const {code, message} = err;
+        expect(code).to.equal(405);
+        expect(message).to.equal('Provider \'rest\' can not call \'remove\'. (disallow)');
+      }
+    });
+  });
+
+  describe('update (put) user', () => {
```

Comment:
Move this describe near to PATH. So, the structure of the tests will be:  POST, GET, PATCH, PUT, DELETE

Reply:
structure changed ed2fcab9e7527ff7434ea9879bb56714970dc4f7